src/app/components/reports/pages/event-summary/EventSummaryReportParameters.ts

```diff
@@ -0,0 +1,221 @@
+  queryStringToNumberArray,
+} from "@helpers/query-string-parameters/query-string-parameters";
+import { Id, Ids } from "@interfaces/apiInterfaces";
+import { ImplementsInjector } from "@models/AbstractModel";
+import { hasMany } from "@models/AssociationDecorators";
+import { bawCollection } from "@models/AttributeDecorators";
+import { AudioEventProvenance } from "@models/AudioEventProvenance";
+import { EventSummaryReport } from "@models/EventSummaryReport";
+import { Region } from "@models/Region";
+import { Site } from "@models/Site";
+import { Tag } from "@models/Tag";
+import { DateTime, Duration } from "luxon";
+
+export enum Chart {
+  speciesAccumulationCurve = "accumulation",
+  speciesCompositionCurve = "composition",
+  falseColorSpectrograms = "false-colour",
+  none = "none",
+}
+
+export enum BucketSize {
+  day = "day",
+  week = "week",
+  fortnight = "fortnight",
+  month = "month",
+  season = "season",
+  year = "year",
+}
+
+const conversionTable = {
+  sites: (value: string) => queryStringToNumberArray(value),
+  points: (value: string) => queryStringToNumberArray(value),
+  provenances: (value: string) => queryStringToNumberArray(value),
+  events: (value: string) => queryStringToNumberArray(value),
+  score: (value: string) => queryStringNumber(value),
+  charts: (value: string) => queryStringArray(value),
+  bucketSize: (value: string) => value,
+  daylightSavings: (value: string) => queryStringBoolean(value),
+  date: (value: string) => queryStringDateTimeArray(value),
+  time: (value: string) => queryStringDurationTimeArray(value),
+};
+
+export interface IEventSummaryReportParameters {
+  sites: Ids | Id[];
+  points: Ids | Id[];
+  provenances: Ids | Id[];
+  tags: Ids | Id[];
+  score: number;
+  bucketSize: BucketSize;
+  daylightSavings: boolean;
+  time: Tuple<Duration, 2>;
+  date: Tuple<DateTime, 2>;
+  charts: Chart[];
+}
+
+export class EventSummaryReportParameters
+  implements IEventSummaryReportParameters, ImplementsInjector
+{
+  public constructor(
+    queryStringParameters: Params = {},
+    public injector?: Injector
+  ) {
+    // since query string parameters are losely typed using a string from the user space
+    // we have to use this hacky implementation of manual key-value checking and assignment
+    Object.entries(queryStringParameters).forEach(([key, value]) => {
+      if (key in conversionTable) {
+        this[key] = conversionTable[key](value);
+      }
+    });
+  }
+
+  // since these properties are exposed to the user in the form of query string parameters
+  // we use the user friendly names
+  @bawCollection()
+  public sites: Ids | Id[];
+  @bawCollection()
+  public points: Ids | Id[];
+  @bawCollection()
+  public provenances: Ids | Id[];
+  @bawCollection()
+  public tags: Ids | Id[];
+  public score: number;
+  public bucketSize: BucketSize = BucketSize.month;
+  public daylightSavings: boolean;
+  public time: Tuple<Duration, 2>;
+  public date: Tuple<DateTime, 2>;
```

Comment:
This form of declaring a tuple isn't very good. Usually, tuples are indeed fixed length arrays, but they typically store different values.

```suggestion
  public time: Tuple<Duration, Duration>;
  public date: Tuple<DateTime, DateTime>;
```

I wonder if we can make a better construct here. Something with a mix/max / lower/upper value. It can still serialize into an array in  QSP.

Reply:
A tuple isn't a TypeScript native type but one we invented as part of the workbench code base. It's _technically_ not a tuple because it only allows one type.

![image](https://github.com/QutEcoacoustics/workbench-client/assets/33742269/7d6a24e4-df57-4c1e-a69d-0220287068ff)

I agree that it'd be really nice if we could specify a min/max value allowed by the tuple, however we can't do this with TypeScript (because TypeScript doesn't support value ranges), so this would have to be done with runtime assertions (probably some sort of decorator with `console.error` (so that unit tests fail on it).

What do you think? I personally think that validating valid value ranges should be done by form validation/at input?

Reply:
https://www.typescriptlang.org/docs/handbook/basic-types.html#tuple

if anything we should change the name of our tuple type to `MonoTuple` or something like that.

as for the rest of your comment, I'm not sure what you mean. I was suggesting a `IExtent` or `Extent` type with  `min` and `max` fields.

Reply:
Sounds good and sounds simple to implement

I'll make the required code changes